app/src/test/kotlin/compiler/parser/analysis/Common.kt

```diff
@@ -0,0 +1,58 @@
+package compiler.parser.analysis
+
+import compiler.common.dfa.DfaWalk
+import compiler.common.dfa.state_dfa.Dfa
+import compiler.common.dfa.state_dfa.DfaState
+import compiler.parser.grammar.Production
+
+typealias R = Production<GrammarSymbol>
+typealias GrammarSymbol = String
+const val EPSILON = ""
```

Comment:
Are we allowed to do something like this? Can our DFA have Epsilons as edges? If yes, then they should be publicly defined as a constant of GrammarSymbol, so that we can omit them in nullable and other functions

Reply:
I removed all EPSILON edges, they weren't really necessary, you may take a look to confirm that these dfas still works correctly.